### PR TITLE
Add tests for playground's dates formatter view

### PR DESCRIPTION
**Pull request by CarlosCortizasCT**
#### Summary

This PR adds RTL tests for playground's dates formatter view.

(_this is a follow up of #2679_)

#### Description

There is a new test which checks every _card_ that gets rendered in the new view with different locales.

### CODE REVIEW

File: playground/src/components/formatters-demo/formatters-demo.jsx [hunk 1]
```diff
@@ -23,9 +25,8 @@ const DEMO_MONEY_PRICE = {
 
 function getSampleDate() {
   const now = new Date();
```
[the remark this answers is not in the document]
Comment (emmenko) [reply playground/src/components/formatters-demo/formatters-demo.jsx:27]: Yes but the VRTs will fail the next month if the UI changes, right?

If we're going to take snapshots, we need to ensure the UI doesn't change. Maybe we don't need VRTs.

We could also test the component that renders the formatted values, so one test for each locale for example. At the end we are interested about the values and not much about the visuals.

[the remark this answers is not in the document]
Comment (CarlosCortizasCT) [reply playground/src/components/formatters-demo/formatters-demo.jsx:27]: Yes, you are right.

So, to be sure what you prefer, the idea is to remove the e2e test and to write from RTL tests for the `LocaleExampleWrapper` component?

[the remark this answers is not in the document]
Comment (emmenko) [reply playground/src/components/formatters-demo/formatters-demo.jsx:27]: Was just an alternative but maybe it's simpler to have RTL tests and to test more variations of the formattings.

I think eventually this stuff will be moved to our "X" library for locales and formatting utils.

[the remark this answers is not in the document]
Comment (CarlosCortizasCT) [reply playground/src/components/formatters-demo/formatters-demo.jsx:28]: I changed the tests so now we have RTL ones.
Please let me know if this is ok. Thanks.

Updated in this commit: 758ac976b361c99b904b3165088c67582fcefafb

File: playground/src/components/formatters-demo/formatters-demo.spec.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+jest.mock('moment/dist/locale/de', () => {});
+jest.mock('moment/dist/locale/de-at', () => {});
+jest.mock('moment/dist/locale/es', () => {});
+jest.mock('moment/dist/locale/es-mx', () => {});
+jest.mock('moment/dist/locale/fr', () => {});
+
+const demoDate = new Date('2022-07-15T07:15:22.000Z');
+const demoTimeZone = 'Europe/Berlin';
+const demoMoney = {
+  centAmount: 1036250,
```
[the remark this answers is not in the document]
Comment (CarlosCortizasCT) [reply playground/src/components/formatters-demo/formatters-demo.spec.js]: It is not actually used. It was a leftover from an initial implementation.

Removed in this commit: 020d7cbca76468be88c24b1b8c816223d32c1cd4

File: playground/src/components/formatters-demo/formatters-demo.spec.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+};
+
+describe('Formatters demo', () => {
+  beforeEach(() => {
+    console.error = jest.fn();
+  });
+
+  it.each(DEMO_LOCALES)(
+    'should render examples based on different locales (%s)',
+    async (locale) => {
```
[the remark this answers is not in the document]
Comment (CarlosCortizasCT) [reply playground/src/components/formatters-demo/formatters-demo.spec.js]: Thanks for the suggestion.

Updated in this commit: 020d7cbca76468be88c24b1b8c816223d32c1cd4

**Review by emmenko — APPROVED** [review]
🥳 Awesome, thanks again for following up on this.

**Pull request merged**